src/main/java/org/gridsuite/network/map/NetworkMapService.java

```diff
@@ -1079,32 +1090,77 @@ public List<VscConverterStationMapData> getVscConverterStations(UUID networkUuid
         }
     }
 
+    public List<HvdcLineMapData> getHvdcLines(UUID networkUuid, String variantId, List<String> substationsId) {
+        Network network = getNetwork(networkUuid, getPreloadingStrategy(substationsId), variantId);
+        if (substationsId == null) {
+            return network.getHvdcLineStream()
+                    .map(NetworkMapService::toMapData).collect(Collectors.toList());
+        } else {
+            Set<HvdcLineMapData> res = new LinkedHashSet<>();
+            substationsId.stream().forEach(id ->
+                    network.getSubstation(id).getVoltageLevelStream().forEach(v ->
```

Comment:
you can do forEach directly without stream

Reply:
Indeed, I didn't pay attention, it was in the old code. I'll correct it.